src/coap_io_lwip.c

```diff
@@ -65,10 +65,10 @@ static void coap_recv(void *arg, struct udp_pcb *upcb, struct pbuf *p, const ip_
   /* this is fatal because due to the short life of the packet, never should there be more than one coap_packet_t required */
   LWIP_ASSERT("Insufficient coap_packet_t resources.", packet != NULL);
   packet->pbuf = p;
-  packet->src.port = port;
-  packet->src.addr = *addr;
-  packet->dst.port = upcb->local_port;
-  packet->dst.addr = *ip_current_dest_addr();
+  packet->addr_info.remote.port = port;
```

Comment:
As `packet` is created by `coap_malloc_type()` which is not guaranteed to set the allocated space zero, the following `coap_endpoint_get_session()` may fail as discussed previously.

Reply:
Good catch. addr_info now preset to 0.